java/java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertToLambdaPreconditionChecker.java

```diff
@@ -137,9 +137,48 @@ private MethodTree getMethodFromFunctionalInterface(TreePath pathToNewClassTree)
                 candidate = (MethodTree)member;
             }
         }
+        // default methods can't be implemented with a lambda
+        ExecutableElement candidateElement = (ExecutableElement) info.getTrees().getElement(new TreePath(pathToNewClassTree, candidate));
+        if (overridesDefaultMethod(candidateElement, (TypeElement) baseElement)) {
+            return null;
+        }
         return candidate;
     }
 
+    private boolean overridesDefaultMethod(ExecutableElement method, TypeElement superType) {
+        Boolean overrides = overridesDefaultMethodImpl(method, superType);
+        return overrides != null && overrides;
+    }
+
+    private Boolean overridesDefaultMethodImpl(ExecutableElement method, TypeElement superType) {
+        List<? extends VariableElement> methodParams = method.getParameters();
+        for (Element e : superType.getEnclosedElements()) {
```

Comment:
I remember looking at the method, but then thinking that the filter step seems unnecessary since the && concatenated checks are both really efficient and also already easy to read. This would add an extra pre-filter-step and we would have to run all the subsequent checks anyway.

I take another look in context with your next suggestion (`Elements.overrides`)

Reply:
switched to `methodsIn` filter